### ISSUE TITLE
Unable to capture Vault from dev agent without VAULT_TOKEN

### ISSUE THREAD

**Issue by davemay99** [hashicorp/hcdiag#79]
I have a Vault dev agent running that does not require a token to capture information.  I can run the commands from the CLI without issue, but `hcdiag` fails with the error `unable to find VAULT_TOKEN`.

Capture should error on an invalid or underprivileged token, but should not require VAULT_TOKEN.

Example capture failure:
```
$ ./hcdiag-darwin-arm64 -consul -vault -nomad -dryrun 
2022-03-04T17:53:27.671-0500 [INFO]  hcdiag: Checking product availability
2022-03-04T17:53:28.535-0500 [ERROR] hcdiag: Failed running Products: error="unable to find VAULT_TOKEN env or ~/.vault-token"
```

Example of working Vault commands:
```
$ printenv | grep VAULT%

$ vault read sys/health -format=json
{
  "request_id": "",
  "lease_id": "",
  "lease_duration": 0,
  "renewable": false,
  "data": null,
  "warnings": null
}
```

**Comment by aliciaclark1066** [comment]
Hi @davemay99, is this still an issue? If so, we'll add it to our backlog.

**Comment by davemay99** [comment]
Yes -- just verified that it still behaves the same way on 0.3.1 as well as `main`.

**Comment by aliciaclark1066** [comment]
@davemay99 We've resolved this issue and will push the fix as part of v0.5.0.
